## Introductory ’omics training

Welcome to the introductory ’omics training short course.

The purpose of this training is to provide a beginners’ bioinformatics
toolbox to analyze ’omics data.

This page will update periodically. Please contact
`pnaderiy [at] bidmc [dot] harvard [dot] edu` for any questions or
suggestions you may have about the codes.

## Tentative topics

### System setup

1.  **Setting Up R and RStudio**

    -   Installation of R, a popular programming language for
        statistical analysis.
    -   Installation of RStudio, a user-friendly integrated development
        environment (IDE) for R.
    -   Basic orientation to the RStudio interface and its features.

2.  **Setting Up Python and Jupyter Notebook**

    -   Installation of Python, a versatile programming language widely
        used in bioinformatics.
    -   Installation of Anaconda, a distribution of Python that includes
        essential data science libraries and package management tools.
    -   Creating and managing Python environments using Conda to isolate
        dependencies.
    -   Introduction to Jupyter Notebook, an interactive web-based
        environment for data analysis.

3.  **Introduction to Bash Scripting**

    -   Overview of the Bash shell and its importance in bioinformatics.
    -   Basic commands and navigation within the command-line interface.
    -   Understanding file systems, directories, and paths in a
        Unix-like environment.

4.  **Introduction to GitHub and Version Control**

    -   Understanding the importance of version control in collaborative
        coding and data analysis.
    -   Introduction to Git, a distributed version control system.
    -   Overview of GitHub, a popular platform for hosting Git
        repositories and collaborating with others.

    \### Basic genomics

5.  **Introduction to Sequencing Technologies**

    -   Overview of RNA-Seq and DNA-Seq technology.
    -   The process of library preparation, sequencing, and data
        acquisition.
    -   Applications in quantifying gene expression and alternative
        splicing.

6.  **Introduction to RNA-Seq for gene expression analysis**

    -   Reading and inspecting RNA-Seq data.
    -   Preliminary QC: outlier detection and normalization.
    -   Basic differetial expression analysis.
    -   Understanding batch effects.

7.  **Functional Annotation and Analysis**

    -   Introduction to gene ontology (GO), pathways, and functional
        annotation terms.
    -   How functional annotation helps in understanding the roles of
        genes.
    -   Resources for accessing functional annotation data.
    -   How pathway analysis tools help in understanding complex
        biological processes.
    -   Examples of pathway analysis in disease research.

8.  **Genomics Databases**

    -   Introduction to databases like Gene Expression Omnibus (GEO) and
        The Cancer Genome Atlas (TCGA).
    -   How to access and retrieve genomics data from these resources.
    -   Real-world examples of studies using publicly available
        datasets.

9.  **Analyzing Whole Genome and/or Exome Sequencing**

    -   Overview of how WGS/WES reveals heterogeneity within cell
        populations.
    -   Applications in developmental biology, immunology, and cancer
        research.

10. **RNA-Seq Module 2**

    -   Small RNA-Seq.
    -   Alternative Splicing.
    -   Modeling complex statistical designs.
    -   Batch effect correction methods.
    -   Unsupervised clustering algorithms.

11. **Functional Analysis Module 2**

    -   Advanced pathway analysis.
    -   Analysis for gene interaction networks.

12. **Sequencing DNA Methylation and Chromatic accessibility**

    -   ATAC-Seq
    -   DNA-Methylation arrays and Bisulfite Sequencing.
    -   Functional analysis of ATAC-Seq and Methylation data.

13. **Single-Cell Genomics**

    -   Overview of single-cell RNA sequencing (scRNA-Seq) and its
        importance.
    -   How scRNA-Seq reveals heterogeneity within cell populations.
    -   Applications in developmental biology, immunology, and cancer
        research.

14. **Integration of Multi-Omics Data**

    -   Explanation of multi-omics approaches (genomics,
        transcriptomics, proteomics, metabolomics).
    -   How integrating different types of omics data provides a
        holistic view of biology.
    -   Challenges and methods for data integration.

15. **Case Studies**

    -   Present real-world examples of genomics studies.

16. **Hands-On Activities**

    -   Provide participants with practical exercises related to gene
        expression analysis or pathway analysis using publicly available
        datasets and bioinformatics tools.

17. **Future Directions**

    -   Discuss emerging trends in functional genomics research, such as
        single-cell multi-omics and spatial transcriptomics.
    -   Consider the potential impact of functional genomics on
        personalized medicine.
